linked_list/linked_list.py

```diff
@@ -9,67 +9,120 @@ def __init__(self, value, next_node = None):
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def length(self):
-        pass
+        current_node = self.head
+        length = 0
+        while current_node != None:
+            length += 1
+            current_node = current_node.next
+        return length
 
     # method that returns the value at a given index in the linked list
     # index count starts at 0
     # returns None if there are fewer nodes in the linked list than the index value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def get_at_index(self, index):
-        pass
+        current_node = self.head
+        if index >= self.length():
+            return None
+        i = 0
+        while current_node != None:
+            if i == index:
+                return current_node.value
+            i += 1
+            current_node = current_node.next
+        return None
 
     # method that returns the value of the last node in the linked list
     # returns None if the linked list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(n)
     def get_last(self):
-        pass
+        current_node = self.head
+        while current_node != None:
+            if current_node.next == None:
+                return current_node.value
+            else:
+                current_node = current_node.next
+        return None
 
     # method that inserts a given value as a new last node in the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(n)
     def add_last(self, value):
-        pass
+        current_node = self.head
+        new_node = Node(value)
+        while current_node != None:
+            if current_node.next == None:
+                current_node.next = new_node
+            current_node = current_node.next
+        else:
+            self.head = new_node
 
     # method to return the max value in the linked list
     # returns the data value and not the node
     def find_max(self):
-        pass
+        max = 0
+        current_node = self.head
+        if current_node == None:
+            return None
+        
+        while current_node:
+            if current_node.value > max:
+                max = current_node.value
+            else:
+                current_node = current_node.next
+        return max
 
     # method to delete the first node found with specified value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def delete(self, value):
-        pass
+        current_node = self.head
+        if current_node.value == value:
+            self.head = current_node.next
+            return 
+        while current_node != None:
+            if current_node.next.value == value:

```

Comment:
If you want to check `current_node.next.value` you should also check to see if `current_node.next` is `None`